completed Home Banner, nav and Why Attend Section

Rubylena opened the pull request:
Closes #109 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced new components: `NavBar`, `Hero`, and `WhyAttend`.
  - Launched a "Coming Soon" page highlighting upcoming conference content.

- **Refactor**
  - Redesigned the main layout with an updated header that integrates site navigation and a refreshed event banner now showcasing the 2025 conference details.
  - Streamlined the presentation on sessions and speakers pages for a more uniform user experience.
  
- **Style**
  - Enhanced typography and responsive design elements to modernize the visual appeal.
  - Added new CSS class for responsive headings.

- **Chores**
  - Made configuration and formatting improvements for overall consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Rubylena committed:
completed Home Banner, nav and Why Attend Section

Rubylena committed:
Fix duplicate ref usage for eventDetailsRef.

Emmanuerl commented on public/hero/api-icon.png:
why do we have the .png and .svg ?

coded1guy replied:
resolved.

Emmanuerl commented on src/assets/why-attend/why-attend-img.png:
same here, why do we have the .svg and .png ?

coded1guy replied:
resolved.

Emmanuerl reviewed (changes requested):
Yours
<img width="220" alt="Screenshot 2025-03-10 at 19 25 51" src="https://github.com/user-attachments/assets/055db0f8-7825-4853-b2aa-8345e6818f97" />

Design
<img width="576" alt="Screenshot 2025-03-10 at 19 28 12" src="https://github.com/user-attachments/assets/5f2c7ea8-62a0-4457-97d1-b8b8a0557ba7" />

Well done @Rubylena , although i noticed the border here isn't as specified in the design


Also the API conf logo on the hero. yours is smaller than the design
Yours
<img width="1800" alt="Screenshot 2025-03-10 at 19 29 50" src="https://github.com/user-attachments/assets/b187a61d-03b4-40b6-9cc9-4b363258f7b5" />

Design
<img width="1127" alt="Screenshot 2025-03-10 at 19 27 15" src="https://github.com/user-attachments/assets/5dc5e8fd-7590-45e8-8e9d-9df959929a15" />

coded1guy committed:
chore: merged 2025 into the branch and resolved conflicts

coded1guy committed:
chore: updated the nav, hero and why attend stuffs
changed the Header component to NavBar and propagated the change respectively
converted single index file folders to single files (NavBar, Hero, WhyAttend)
made some changes to fit the design
made necessary change suggested by coderabbit and made changes requested by Kruse

coded1guy committed:
chore: display the footer
I hid the footer while trying to figure out what section was overflowing and pushing the page away from the right, I've reverted that
